mongodbatlas/provider_test.go

```diff
@@ -83,6 +83,13 @@ func testAccPreCheckBasicOwnerID(tb testing.TB) {
 	}
 }
 
+func testAccPreCheckAtlasUsername(tb testing.TB) {
+	testAccPreCheckBasic(tb)
+	if os.Getenv("MONGODB_ATLAS_USERNAME_CLOUD_DEV") == "" {
+		tb.Fatal("`MONGODB_ATLAS_USERNAME_CLOUD_DEV` must be set ")
+	}
+}
```

Comment:
do you plan to add this env var to the repo? if yes, what is going to happen when two PRs update files in the `config` job? I think the acceptance tests may be flaky in this scenario

Reply:
This env variable was already defined in the repo so no changes needed in that regard.
Regarding flakyness, the username is being used for both data source tests:
- `atlas_user` data source: uses the username to fetch the user details.
- `atlas_users` data source: make use of the username for creating a new project/team having that user included, and then the `atlas_users` data source is used to fetch users in the created project/team.

The only risk concern with flakyness as see is with the `atlas_user` data source which is being addressed in this comment: https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1432#discussion_r1311706144